src/fairseq2/recipes/eval/__init__.py

```diff
@@ -4,19 +4,23 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+from pathlib import Path
+from typing import Any, Callable
+
 from fairseq2.logging import get_log_writer
 from fairseq2.recipes.cli import Cli, CliGroup, RecipeCommandHandler
-from fairseq2.recipes.eval.configs import hf_presets
+from fairseq2.recipes.eval.asr import AsrEvalConfig
+from fairseq2.recipes.eval.configs import wav2vec2_presets, whisper_presets
 
 log = get_log_writer(__name__)
 
 
 def _add_wav2vev2_asr_eval_cli(group: CliGroup) -> None:
-    from fairseq2.recipes.eval.asr import load_wav2vec2_asr_evaluator
+    from fairseq2.recipes.eval.asr import ASREvaluator
```

Comment:
I don't know if changing the function "load_wav2vec2_asr_evaluator" to ASREvaluator is the best way. I'm not picky between having a function or a callable class, but the problem is that Whisper is an end-to-end model, while wav2vec2 is - as the name suggests - only an encoder that generates a vector. For wav2vec2, we need a text tokenizer and decoder, while for Whisper it is not required. So the "ASREvaluator" is still not abstract enough (at least your currently proposal, with `self.tokenizer` and `self.decoder`)

Basically I think we need just 2 functions to generate the HFEvaluator accordingly from its config (see my comments below)

Reply:
This makes lots of sense. I was just afraid of having two many functions just to support different models and thier requirements and that's why I created the class.